otherlibs/stdune/fpath.ml

```diff
@@ -31,12 +31,17 @@ let rec mkdir_p ?(perms = 0o777) t_s =
         []
     else
       let parent = Filename.dirname t_s in
-      match mkdir_p ~perms parent with
-      | Created | Already_exists ->
+      match mkdir_p ?perms parent with
+      | Created | Already_exists -> (
         (* The [Already_exists] case might happen if some other process managed
            to create the parent directory concurrently. *)
-        Unix.mkdir t_s perms;
-        Created)
+        match mkdir t_s ?perms with
+        | Created | Already_exists -> Created
```

Comment:
It's odd that we return `Created` if we hit both `Already_exists` branches. It probably doesn't matter much but it may be confusing if two concurrent calls to `mkdir_p` on the same path succeed with `Created`.

Reply:
Good point. I'll return the value from `mkdir`